src/discussion/reaction_models.py

```diff
@@ -51,7 +51,10 @@ def __str__(self):
 
     @property
     def paper(self):
-        return self.item.paper
+        from paper.related_models.paper_model import Paper
+
+        if isinstance(self.item, Paper):
+            return self.item
```

Comment:
I think we should remove this property. `instance.item` is good enough.

Reply:
@calvinhlee23

Reply:
No. `@paper` should never return any other item type such as Hypothesis, Posts, Comments when the item is still present.